lib/config_skeleton.rb

```diff
@@ -242,11 +242,16 @@ def run
     logger.debug(logloc) { "notifier fd is #{notifier.to_io.inspect}" }
 
     loop do
-      if ios = IO.select(
-          [notifier.to_io, @terminate_r, @trigger_regen_r],
-          [], [],
-          sleep_duration.tap { |d| logger.debug(logloc) { "Sleeping for #{d} seconds" } }
-      )
+      if cooldown_duration > 0
+        logger.debug(logloc) { "Sleeping for #{cooldown_duration} seconds (cooldown)" }
+        IO.select([@terminate_r], [], [], cooldown_duration)
+      end
+
+      timeout = sleep_duration - cooldown_duration
```

Comment:
Perhaps we should have validation on startup that cooldown_duration is < sleep_duration and throw an error if it is not; we are just trusting the developer here to do the right thing. Like what happens if timeout gets passed a negative number to IO.select?